### PR TITLE
Fix multiple typos

### PR DESCRIPTION
**Pull request by JonHanna**

### CODE REVIEW

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/MemberLookup.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ internal class MemberLookup
         // These are for error reporting.
         private SymWithType _swtAmbig;     // An ambiguous symbol.
         private SymWithType _swtInaccess;  // An inaccessible symbol.
-        private SymWithType _swtBad;       // If we're looking for a ructor or indexer, this matched on name, but isn't the right thing.
+        private SymWithType _swtBad;       // If we're looking for a constructor or indexer, this matched on name, but isn't the right thing.
```
Comment (JonHanna) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/MemberLookup.cs]: When find-and-replace attacks!

**Comment by JonHanna** [comment]
Test  Innerloop OSX Release Build and Test

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/MemberLookup.cs (hunk 1 above)
Comment (stephentoub) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/MemberLookup.cs]: :)

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/AggregateType.cs [hunk 2]
```diff
@@ -29,8 +29,8 @@ internal partial class AggregateType : CType
         public short typeConverterID;
 #endif // !CSEE
 
-        public bool fConstraintsChecked;    // Have theraints been checked yet?
-        public bool fConstraintError;       // Did theraints check produce an error?
+        public bool fConstraintsChecked;    // Have the constraints been checked yet?
+        public bool fConstraintError;       // Did the constraints check produce an error?
```
Comment (stephentoub) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/AggregateType.cs]: It is interesting... I wonder in what situation in the past someone did a search and replace to remove all "const".

**Comment by stephentoub** [comment]
LGTM

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/AggregateType.cs (hunk 2 above)
Comment (JonHanna) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/AggregateType.cs]: I'm pretty sure they did. And then it compiled fine, so it must have been okay :smile:

Comment (JonHanna) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/AggregateType.cs]: Oh wait. Yes, when would that find-replace ever make sense in C#? Curious indeed.

**Pull request merged**